### PR TITLE
upgrade to v2.5.4

### PR DESCRIPTION
**Pull request by cdeepali**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -53,10 +53,19 @@ cp make.inc-examples/make.inc.openblas ./make.inc
 # standard supported by both tool sets.
 export CXXFLAGS="$(echo $CXXFLAGS | sed -e 's/ -std=[^ ]*//') -std=c++14"
 
+GPU_TARGET="sm_37 sm_60 sm_70 sm_75"
```
Comment (jayfurmanek) [inline recipe/build.sh:56]: I eventually want to pull this out to a central location, but for now lets standardize on:

CUDA10.2:
x86:
    CUDA_OPTION_1='sm_37,sm_52,sm_60,sm_61,sm_70,compute_75'

ppc64le:
    CUDA_OPTION_1='sm_37,sm_60,sm_70,compute_75'

CUDA11:
x86:
    CUDA_OPTION_1='sm_37,sm_52,sm_60,sm_61,sm_70,sm_75, compute_80'

ppc64le:
    CUDA_OPTION_1='sm_37,sm_60,sm_70,sm_75, compute_80'

**Review by jayfurmanek — COMMENTED** [review]
This looks good. Let's wait to merge until we are ready to push in PyTorch 1.7 updates.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 package:
   name: magma
   version: {{ version }} 
 
 source:
    url: http://icl.utk.edu/projectsfiles/magma/downloads/magma-{{ version }}.tar.gz
-   sha1: 809867a477d67d0b137f6b2cea4eecaec057f53c 
+   sha1: c428c9486c5f46f3595570ec898274a8bb888d5c
    patches:
      - cmakelists.patch
```
Comment (jayfurmanek) [inline recipe/meta.yaml:11]: Let's organize these patches like we have for some of the other recipes (like tensorflow) into categories.

File: recipe/build.sh (hunk 1 above)
Comment (cdeepali) [reply recipe/build.sh:56]: magma is internally expanding supplied `sm_*` flags to:

`-- CUDA_NVCC_FLAGS: -DHAVE_CUBLAS;-Xfatbin;-compress-all;-Xcompiler;-fPIC;-std=c++11;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_80,code=compute_80;-DADD_`

**Review by bnemanich — APPROVED** [review]

**Pull request merged**